src/physics_common.cpp

```diff
@@ -23,4 +24,160 @@ void russian_roulette(Particle* p)
+  else 
+    num_splits = int(std::ceil(importance_ratio));
+  // calculate the tru number of splits
+  p->wgt_ /= float(num_splits);
+  // note 1 -> n not 0->n
+  for ( int i = 1 ; i < num_splits ; i++ ) {
+    p->create_secondary(p->u(),p->E_,p->type_);
+  }
+
+  return;
+}
+
+//==============================================================================
+// ROULETTE
+//==============================================================================
+
+void roulette(Particle* p, double importance_ratio) {
+  // this is a kill event
+  double random = prn();
+  //std::cout << random << " " << importance_ratio;
+  if ( random < importance_ratio) {
+    //std::cout << " survive " << std::endl;
+    p->wgt_last_ = p->wgt_;
+    //std::cout << p->last_wgt << " ";
+    p->wgt_ /= importance_ratio;
+     //std::cout << p->wgt << std::endl;  
+  } else {
+    //std::cout << " kill" << std::endl;
+    p->alive_ = false;
+    p->wgt_ = 0.;
+    p->wgt_last_ = 0.;
+  }
+  return;
+}
+
+//==============================================================================
+// WEIGHT_WINDOW - assume we have already checked cutoffs
+//==============================================================================
+
+void weight_window_split(Particle *p) {
+  bool mesh_weight = false;
+  double weight_low = 0.;
+  double weight_high = 0.;
+
+  /* - todo complete weight from mesh
+  if ( mesh_weight ) 
+    get_window_from_mesh(p,weight_low,weight_high);
+  else
+    get_window_from_cell(p,weight_low,weight_high);
+  */
+  if ( p->wgt_ > weight_high ) {
+    if ( p->wgt_/weight_high > MAX_SPLIT ) {
+          // long history mitigation - if the number of requested
+          // splits is greater than the prescribed amount in the input
+          // lets say a million then we increase the effective weight
+          // map of the problem until this history ends
+
+          // todo - dynamic sharing of the split history oversplitting
+          // is an issue mostly because of the parallel efficiency drop
+          // perhaps this seconday particle can be distributed over the
+          // source bank instead? need to be sure the random number state
+          // isnt peturbed      
+    }
+    /* 
+    // split into n secondaries
+    int n_secondary = std::floor(p->wgt/weight_high);
+    double secondary_wgt = p->wgt/n_secondary;
+    double remaining_wgt = p->wgt - n_secondary*secondary_wgt;
+  
+    p->wgt = secondary_wgt;
+    // create secondaries with the correct weight
+    for ( int i = 1 ; i < n_secondary) {
+      p->create_secondary(p->uvw,p->E,p->type,true);
+    }
+    // reset original particle to include left over weight
+    p->wgt += remaining_wgt;
+    */
+    split_by_ratio(p,p->wgt_/weight_high);
+  } else if ( p->wgt_ < weight_low ) {
+    roulette(p,p->wgt_/weight_low);  
+  } else {
+    // do nothing we are in the window
+  }
+  return;  
+}
+
+// split according to importance 
+void importance_split(Particle *p){
```

Comment:
Anticipating other possible sources of importance info?
```suggestion
void cell_importance_split(Particle *p){
```